crypto/evp/evp_pbe.c

```diff
@@ -25,41 +27,42 @@ struct evp_pbe_st {
     int cipher_nid;
     int md_nid;
     EVP_PBE_KEYGEN *keygen;
+    EVP_PBE_KEYGEN_EX *keygen_ex;
 };
 
 static STACK_OF(EVP_PBE_CTL) *pbe_algs;
 
 static const EVP_PBE_CTL builtin_pbe[] = {
     {EVP_PBE_TYPE_OUTER, NID_pbeWithMD2AndDES_CBC,
-     NID_des_cbc, NID_md2, PKCS5_PBE_keyivgen},
+     NID_des_cbc, NID_md2, PKCS5_PBE_keyivgen, &PKCS5_PBE_keyivgen_ex},
```

Comment:
I wonder if the PBE stuff should become something a provider offers.  I.e. a new algorithm type.

Reply:
Sounds like a good idea until you try doing it :)
EVP_PBE_CipherInit() is using too much ASN!_OBJECT stuff.. It ends up just doing a KDF which we already have in a provider and then using the derived key/iv to pass to a cipher.. So not much gain for the pain..

Reply:
#14326 was done to shift one of the KDF's that is used by the keygen...

Reply:
Yup, I started down that path to see what would happen but it's messy. The ASN.1 encode/decode stuff is intertwined and would need new decoders. Significantly more work IMO.